screens/Scan/ScanScreenController.ts

```diff
@@ -31,20 +34,22 @@ export function useScanScreen({ navigation }: MainRouteProps) {
   const isFlightMode = useSelector(scanService, selectIsAirplaneEnabled);
 
   const locationError = { message: '', button: '' };
-  if (isFlightMode) {
-    locationError.message = t('errors.flightMode.message');
-    locationError.button = t('errors.flightMode.button');
-  } else {
-    if (isLocationDisabled) {
-      locationError.message = t('errors.locationDisabled.message');
-      locationError.button = t('errors.locationDisabled.button');
-    } else if (isLocationDenied) {
-      locationError.message = t('errors.locationDenied.message');
-      locationError.button = t('errors.locationDenied.button');
-    }
-  }
 
   useEffect(() => {
+    if (isFlightMode) {
+      locationError.message = t('errors.flightMode.message');
+      locationError.button = t('errors.flightMode.button');
+      return;
+    } else {
+      if (isLocationDisabled) {
+        locationError.message = t('errors.locationDisabled.message');
+        locationError.button = t('errors.locationDisabled.button');
+      } else if (isLocationDenied) {
+        locationError.message = t('errors.locationDenied.message');
+        locationError.button = t('errors.locationDenied.button');
+      }
+    }
+
```

Comment:
Moving it here means the location error will no longer update after the first render. Is this what you intended?

Reply:
@pmigueld I moved this because it gives lint error